### PR TITLE
fix(avatars): resize avatars before cropping them

### LINKED ISSUE #1916: Avatar cropper: rotating a large image causes OSX beachball and huge memory usage spike
STR

1) Signup & verify
2) Go to /settings/avatar
3) Click "Change"
4) Click "Upload"
5) Upload a large image (mine was 6000x4000)
6) Open the OS' Activity Monitor (or similar tool)
7) Rotate the image
8) Watch the memory spike.

My Firefox memory footprint went from 1.1GB to 2.5GB, and I saw the OSX beachball for about 45 seconds.

### PR DESCRIPTION
**Pull request by eoger**
Fixes #1916

### CODE REVIEW

**Comment by eoger** [comment]
No more huge rotating lag, but comes at the price of resizing first.

**Comment by vladikoff** [comment]
```
{"stat":"scrypt","maxPending":0,"numPending":0,"numPendingHWM":0}}
Failed test: resize
 => AssertionError: expected 0 to equal 100
Failed test: resize handles over 100 value
 => AssertionError: expected 0 to equal 100
× firefox on any platform - mocha tests - run the mocha tests (37.702s)
Error: Expected 0 mocha test failures
```

**Comment by vladikoff** [comment]
@zaach do you have time to review this?

**Comment by zaach** [comment]
Looks like it needs a unit test for the new resize function and the new constructor behavior.

**Comment by eoger** [comment]
Updated!

File: app/tests/spec/lib/cropper.js [hunk 1]
```diff
@@ -188,27 +206,35 @@ function (chai, sinon, jQuerySimulate, RouterMock, CanvasMock, p, Cropper, Ephem
       });
 
       it('resize', function () {
+        var img = { width: 100, height: 50, src: pngSrc };
+        var resized = cropper.resize(img, 0.5);
+        assert.equal(resized.src, 'data:image/jpeg');
```
Comment (zaach) [inline app/tests/spec/lib/cropper.js]: `canvas.toDataURL` should return type `data:image/png` [by default](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) and `resize` isn't passing any arguments, so not sure why this is returning `jpeg`.

Comment (eoger) [inline app/tests/spec/lib/cropper.js]: It's a mock canvas :)

Comment (zaach) [inline app/tests/spec/lib/cropper.js]: Ah, okay. Not worth delaying the train over.

**Pull request merged**